### PR TITLE
Use Python 3.6 for `cupy_install` test

### PR DESCRIPTION
**Pull request by emcastillo**
USes py36-venv in ubuntu 16 because of ubuntu 18 and cuda 9.0 c compiler issues

### CODE REVIEW

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Closes #616

**Comment by emcastillo** [comment]
Jenkins, test this please

File: run_cupy_install_test.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
     # make sdist
     # cuda, cudnn and numpy is required to make a sdist file.
     build_conf = {
-        'base': 'ubuntu16_py35',
+        'base': 'ubuntu16_py36-pyenv',
```
Comment (asi1024) [inline run_cupy_install_test.py:44]: How about using ubuntu18? The EOL of ubuntu16 (2021-04) is coming soon.
```suggestion
        'base': 'ubuntu18_py36',
```

Comment (emcastillo) [reply run_cupy_install_test.py:44]: Yeah I also tried that, but the problem is that we need to install an older gcc for cuda 9.0 to work :(.
I wonder if we can bump the cuda version for this test? @kmaehashi

Comment (asi1024) [reply run_cupy_install_test.py:44]: (internal discussion) We decided to bump cuda version for the install test.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**